configs/templates/unified-dev/spack.yaml

```
@@ -0,0 +1,32 @@
+spack:
+  concretizer:
+    unify: when_possible
+
+  view: false
+  include: []
+
+  definitions:
+  - compilers: ['%apple-clang', '%gcc', '%intel']
```

```suggestion
  - compilers: ['%apple-clang', '%gcc', '%intel']
  #- compilers: ['%apple-clang']                # for the Mac only
```

I had to delete '%gcc' and '%intel' like in the suggested code to get this to work on my Mac. I think that is okay - the Mac is probably the outlier here. I'm okay if you have a better way to accomplish this.

---

I don't want to do this, honestly. I am thinking to update the documentation with either instructions on how to manually change this (everyone needs to, HPCs and Linux systems only have gcc and intel, but not apple-clang), or make automate this as part of the `spack stack create env` or a separate command.

---

@climbfuji I'm okay with the solutions you are suggesting. I agree these are better than placing comments in the template scripts. Updating the documentation is probably more expedient, but it's your call.

---

@climbfuji Is it possible we could kill two birds with one stone re: https://github.com/NOAA-EMC/spack/pull/213? Maybe for each site, have a list of compiler/MPI combinations (with versions explicitly included) that `spack stack create env` can use to populate that part of spack.yaml? I don't see any way to have compilers listed in, say, site/spack.yaml but then have packages listed in a separate file...

---

Yes, that would be great to have.

---

I have updated the documentation to manually adjust this line, we can make @AlexanderRichert-NOAA's changes in a follow-up PR to automate the change of that line in the spack config.

---

Thanks for updating the documentation!